anndata/tests/test_readwrite.py

```diff
@@ -619,6 +620,24 @@ def test_write_string_types(tmp_path, diskfmt):
         write(adata_pth)
 
 
+@pytest.mark.parametrize(
+    "teststring", ["teststring", np.asarray(["test1", "test2", "test3"])]
```

Comment:
What `dtype` are you expecting `np.asarray(["test1", "test2", "test3"])` to be read back in as?

Reply:
depending on how it was written, it can be either `dtype=object` (for variable-length UTF-8 and ASCII strings, in the latter case the array will contain bytestring objects) or `dtype=S` for fixed-length strings

Reply:
Even if it's ascii, I would like it to be read in as a string. It could be a string encoded as ascii, but it would be equivalent to read it encoded as utf-8.

My main concern with this exact test is that `np.asarray(["test1", "test2", "test3"])` creates a numpy's unicode strings, which h5py (and hdf5) doesn't support. I think it would make sense for this to be more like:

```suggestion
    "teststring", ["teststring", np.asarray(["test1", "test2", "test3"], dtype=object)]
```